plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/LdapGroupProvider.java

```diff
@@ -93,6 +94,18 @@ public LdapGroupProvider(LdapClient client, LdapGroupProviderConfig ldapConfig)
                 CacheBuilder.newBuilder()
                         .expireAfterWrite(ldapConfig.getLdapCacheTtl().toMillis(), MILLISECONDS),
                 CacheLoader.from(this::getGroupsWithBindDistinguishedName));
+        if (this.groupUserMembershipAttribute.isPresent()) {
+            this.ldapGroupProviderQueryUser = Optional.of(new LdapGroupProviderQueryUser());
+        }
+        else {
+            this.ldapGroupProviderQueryUser = Optional.empty();
+        }
+        if (this.groupMembershipAttribute.isPresent()) {
+            this.ldapGroupProviderQueryGroup = Optional.of(new LdapGroupProviderQueryGroup());
+        }
+        else {
+            this.ldapGroupProviderQueryGroup = Optional.empty();
+        }
```

Comment:
We can inject `LdapGroupProviderQuery` implementation based on the config property so we don't have to maintain two variables one for `ldapGroupProviderQueryUser` and `ldapGroupProviderQueryGroup`

Reply:
Please check the usage of `AbstractConfigurationAwareModule` for specifying which implementation to bind.